src-electron/db/query-device-type.js

```diff
@@ -661,6 +661,56 @@ async function selectDeviceTypeFeaturesByEndpointTypeIdAndClusterId(
   return rows.map(dbMapping.map.endpointTypeDeviceExtended)
 }
 
+/**
+ * Retrieves the endpoint composition requirements for a given device type.
+ * Returns all required device types that must be on separate endpoints, along with
+ * their conformance and constraint information.
+ *
+ * Example: For an Oven device type, this would return that it requires
+ * at least 1 Temperature Controlled Cabinet device type on a separate endpoint.
+ *
+ * @param {Object} db - The database connection object.
+ * @param {number} deviceTypeRef - The device type reference ID (DEVICE_TYPE_ID).
+ * @returns {Promise<Array>} A promise that resolves with an array of required device types with their constraints.
+ */
+async function selectEndpointCompositionRequirementsByDeviceTypeRef(
+  db,
+  deviceTypeRef
+) {
+  // Single query that joins ENDPOINT_COMPOSITION and DEVICE_COMPOSITION
+  // using DEVICE_TYPE_REF directly, avoiding the need for two separate queries
+  const query = `
+    SELECT
+      DC.CONFORMANCE,
+      DC.DEVICE_CONSTRAINT,
+      DC.DEVICE_TYPE_REF as required_device_type_ref,
+      DT_REQ.NAME as required_device_name,
```

Comment:
We generally have these all upper case. Why do you have the "as ...." in lowercase?